src/journal/managed-journal-file.c

```diff
@@ -512,19 +512,25 @@ int managed_journal_file_open_reliably(
         if ((flags & O_ACCMODE) == O_RDONLY)
                 return r;
 
-        if (!(flags & O_CREAT))
-                return r;
-
         if (!endswith(fname, ".journal"))
                 return r;
 
         /* The file is corrupted. Rotate it away and try it again (but only once) */
-        log_warning_errno(r, "File %s corrupted or uncleanly shut down, renaming and replacing.", fname);
+        log_warning_errno(r, "File %s corrupted or uncleanly shut down, trying to move it out of the way: %m", fname);
 
         r = journal_file_dispose(AT_FDCWD, fname);
-        if (r < 0)
+        if (r < 0) {
+                log_warning_errno(r, "Failed to move %s out of the way: %m", fname);
```

Comment:
if you propagate the error and log here this means we'll log twice here. that sucks.

Reply:
The 4th patch should remove the redundancy, hence when `vacuum_offline_user_journals()` fails we should see:

```
File system.journal corrupted or uncleanly shut down, trying to move it out of the way: %m
Failed to move system.journal out of the way: %m
Failed to read journal file system.journal for rotation, ignoring: %m
```
which seems OK, no ?